src/node_http_parser.cc

```diff
@@ -636,6 +644,24 @@ class Parser : public AsyncWrap, public StreamListener {
     if (ret.IsEmpty())
       return;
 
+    // check header parsing time
+    if (header_parsing_start_time_ != 0 && headers_timeout_ != 0) {
+      auto now = uv_hrtime();
+      auto parsing_time = (now - header_parsing_start_time_) / 1e6;
+
+      if (parsing_time > headers_timeout_) {
+        Local<Value> cb =
```

Comment:
is there a good way to avoid having an extra callback here?

Reply:
@orkon I don’t think so. But if it only fires in the timeout case, that should be okay.